### ISSUE TITLE
finally clauses aren't treated properly in branch coverage?

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#92]
Brodie Rao measured cram:  http://bitheap.org/cram/htmlcov/cram.html

The finally clauses are marked as partial, because they exited normally, rather than with exceptions.  And the missed exit is marked as the loop exit rather than the function exit, why is that?

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/92
- This issue had attachments: cram.png.  See the [original issue](https://bitbucket.org/ned/coveragepy/issue/92) for details.

**Comment by nedbat** [comment]
A simpler test case:

```
                for i in range(5):
                    try:
                        j = 3
                    finally:
                        f = 5
                    g = 6
                h = 7
```

**Comment by nedbat** [comment]
Fixed in changeset [2c5fb3a8b81c (bb)](https://bitbucket.org/ned/coveragepy/commits/2c5fb3a8b81c).
